lib/plugins/aws/package/compile/events/msk/index.test.js

```diff
@@ -0,0 +1,242 @@
+        ],
+        Resource: '*',
+      });
+    });
+
+    it('should correctly compile EventSourceMapping resource DependsOn ', () => {
+      expect(eventSourceMappingResource.DependsOn).to.equal('IamRoleLambdaExecution');
+    });
+  });
+
+  describe('when using all parameters', () => {
+    it('should correctly compile EventSourceMapping resource', () => {
+      const enabled = false;
+      const startingPosition = 'LATEST';
+      const batchSize = 5000;
+
+      return fixtures
+        .extend('function', {
+          functions: {
+            foo: {
+              events: [
+                {
+                  msk: {
+                    topic,
+                    arn,
+                    batchSize,
+                    enabled,
+                    startingPosition,
+                  },
+                },
+              ],
+            },
+          },
+        })
+        .then(fixturePath =>
+          runServerless({
+            cwd: fixturePath,
+            cliArgs: ['package'],
+          }).then(({ awsNaming, cfTemplate }) => {
+            const resource =
+              cfTemplate.Resources[
+                awsNaming.getMSKEventLogicalId('foo', 'ClusterName', 'TestingTopic')
+              ];
+            expect(resource.Properties).to.deep.equal({
+              BatchSize: batchSize,
+              Enabled: enabled,
+              EventSourceArn: arn,
+              StartingPosition: startingPosition,
+              Topics: [topic],
+              FunctionName: {
+                'Fn::GetAtt': [awsNaming.getLambdaLogicalId('foo'), 'Arn'],
+              },
+            });
+          })
+        );
+    });
+  });
+
+  describe('when no msk events are defined', () => {
+    it('should not modify the default IAM role', () => {
+      return runServerless({
+        cwd: fixtures.map.function,
+        cliArgs: ['package'],
+      }).then(({ cfTemplate }) => {
+        const defaultIamRole = cfTemplate.Resources.IamRoleLambdaExecution;
+        expect(defaultIamRole.Properties.Policies[0].PolicyDocument.Statement).not.to.deep.include({
+          Effect: 'Allow',
+          Action: ['kafka:DescribeCluster', 'kafka:GetBootstrapBrokers'],
+          Resource: [],
+        });
+
+        expect(defaultIamRole.Properties.Policies[0].PolicyDocument.Statement).not.to.deep.include({
+          Effect: 'Allow',
+          Action: [
+            'ec2:CreateNetworkInterface',
+            'ec2:DescribeNetworkInterfaces',
+            'ec2:DescribeVpcs',
+            'ec2:DeleteNetworkInterface',
+            'ec2:DescribeSubnets',
+            'ec2:DescribeSecurityGroups',
+          ],
+          Resource: '*',
+        });
+      });
+    });
+  });
+});
+
+describe('getMSKClusterName', () => {
+  let awsCompileMSKEvents;
+
+  before(() => {
+    const serverless = new Serverless();
+    serverless.setProvider('aws', new AwsProvider(serverless));
+    awsCompileMSKEvents = new AwsCompileMSKEvents(serverless);
```

Comment:
Let's not test it like that, as it may be problematic for eventual internal refactors.

I'd rather introduce all three cases in `runServerless` tests, and just confirm the outcome is expected. (that should be good enough)

Eventually we may  seclude `getMSKClusterName` to standalone util (e.g. to `getMSKClusterName.js` in same folder) and write unit tests for that

Reply:
I went with moving it out to separate file as I feel it's a bit "cleaner" to unit test this. Please let me know what do you think about such approach.